cmds/modules/noded/main.go

```diff
@@ -202,6 +205,41 @@ func action(cli *cli.Context) error {
 		return errors.Wrap(err, "failed to create a new perfMon")
 	}
 
+	if exists := iperf.Exists(zinit.Default()); exists {
+		g, err := graphql.NewGraphQl(env.GraphQL)
+		if err != nil {
+			return errors.Wrap(err, "failed to create a new graphql")
+		}
+
+		freeFarmNodes, err := g.ListPublicNodes(0, 1, true, false)
+		if err != nil {
+			return errors.Wrap(err, "failed to list freefarm nodes from graphql")
+		}
+
+		nodes, err := g.ListPublicNodes(12, 0, true, false)
+		if err != nil {
+			return errors.Wrap(err, "failed to list nodes from graphql")
+		}
+
```

Comment:
This is too much to be done by noded. Noded all what it cares about is to register a certain test against the perf framework. 

The test ON RUNNING, will pick the nodes to test against. means the nodes can be different each time the test runs